server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -154,21 +155,23 @@ public void validate(IndexSettings settings, boolean checkLimits) {
                 }
             }
         }
-        List<String> routingPaths = settings.getIndexMetadata().getRoutingPaths();
-        for (String path : routingPaths) {
-            for (String match : mappingLookup.getMatchingFieldNames(path)) {
-                mappingLookup.getFieldType(match).validateMatchedRoutingPath(path);
-            }
-            for (String objectName : mappingLookup.objectMappers().keySet()) {
-                // object type is not allowed in the routing paths
-                if (path.equals(objectName)) {
-                    throw new IllegalArgumentException(
-                        "All fields that match routing_path must be configured with [time_series_dimension: true] "
-                            + "or flattened fields with a list of dimensions in [time_series_dimensions] "
-                            + "and without the [script] parameter. ["
-                            + objectName
-                            + "] was [object]."
-                    );
+        if (settings.getMode() == IndexMode.TIME_SERIES) {
```

Comment:
Would this also be the case of logsdb?

Reply:
The former loop is for dimensions so not applicable, restored the second one and updated the message.